WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Categories/ProductCategoryListViewModelTests.swift

```diff
@@ -34,10 +36,139 @@ final class ProductCategoryListViewModelTests: XCTestCase {
         // Then
         XCTAssertFalse(isCategorySelected)
     }
+
+    func testSyncedStateForASinglePageCategoryList() {
+        // Given
+        let exp = expectation(description: #function)
+        let product = MockProduct().product()
+        let viewModel = ProductCategoryListViewModel(product: product)
+        let storesManager = MockProductCategoryStoresManager()
+        storesManager.productCategoryResponses = [(sampleCategories(count: 3), nil), ([], nil)]
+        ServiceLocator.setStores(storesManager)
```

Comment:
Exactly, I wanted to do that, but didn't found(maybe didn't looked well enough) a place where dependencies that come from `ServiceLocator` were injected in the initializer. So I choose to be aligned with the current style of the project.

I'm more than happy on changing this tho, I'm a strong supporter of DI as you could see on my post earlier 😅

Reply:
We have it in a few places:

https://github.com/woocommerce/woocommerce-ios/blob/a578b8808833a45222edbf762ccab8fe3e60b8da/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/DefaultProductFormTableViewModel.swift#L18-L21

We're _slowly_ moving towards it. There are just a few roadblocks in terms of unit testing (e.g. https://github.com/woocommerce/woocommerce-ios/issues/1765) that somehow encourages us not to use the pattern because we couldn't test the `ViewModel` anyway. 🤷 

But if the `ViewModel` can be tested (and it should be testable), we should use this pattern. 

So, yes, please proceed. 😁 

--- 

On a related note, we'd also like the _inject_ the `siteID` so it's constant. We don't want that changing under our noses.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/2130/commits/1163c9801f6fb964f3cc15f6aa1c76ffcfcd5f18